### ISSUE TITLE
some data types aren't being stored in the global metadata

### ISSUE THREAD

**Issue by mrRay** [Synopsis/Synopsis#54]
for example, motion isn't presently being stored in the global metadata

**Comment by vade** [comment]
Thanks @mrRay - as per our chat, some stuff hasn't been fully implemented yet so some keys are not in global yet.

However, upon pondering this a bit, its unclear if it makes sense to try to have every per frame analysis pass have an entry in the global metadata, or at least, have an entry that exactly matches what a per frame output will have.

Some gotchas off the top of my head:

* Some metadata simply might not exist per frame (say tracking points)
* Some metadata might not be easy to summarize globally in the same way - might be able to say something about on average how many tracking points, but I cant necessarily summarize every point.
* Some global metadata might just outright not exist per frame in the future (data that only makes sense to describe the entirety of the frames rather than one frame in particular).

I think you mentioned you wanted some way to know 'what keys' to expect for per frame metadata. Is that correct? 

How fucked would things be if you simply could not guarantee the decoded JSON has a key every frame?

**Comment by mrRay** [comment]
> I think you mentioned you wanted some way to know 'what keys' to expect for per frame metadata. Is that correct?

yes, sort of- i want to know what analysis keys i should expect during playback.  when i trigger a clip, i have to set up some stuff on the backend to support publishing of the various metadata values.  in order to set this stuff up, i need to know exactly what kinds of values i'm going to be encountering (if i don't know this then i'll have to do this backend setup on the fly, when the metadata is received- and that's non-optimal).

when i filed this bug report, i was literally just looking for a way to programmatically extract a list of the attributes that are stored with a given video.  like, all i was looking for here is something like ("AverageColor","DominantColors","Motion").

...uhh, if anyone else is reading this and they have a similar request: you can work around this by making an AVSampleBufferGenerator for the metadata track and request metadata blobs as sample buffer refs.

**Comment by vade** [comment]
@mrRay - that still won't guarantee that you get 'every possible output' from the first sample buffer of metadata though. We designed the metadata to have some sparse items since there are sometimes only momentary data.
